src/stratis_cli/_actions/_top.py

```diff
@@ -538,3 +602,139 @@ def add_cache_devices(namespace):  # pylint: disable=too-many-locals
                 )
                 % (namespace.pool_name, ", ".join(devnodes_added), ", ".join(blockdevs))
             )
+
+    @staticmethod
+    def add_key(namespace):
+        """
+        Add key to kernel keyring.
+
+        :raises StratisCliEngineError:
+        :raises StratisCliPropertyNotFoundError:
+        :raises StratisCliNameConflictError:
+        :raises StratisCliIncoherenceError:
+        """
+        proxy = get_object(TOP_OBJECT)
+
+        key_list = fetch_keylist_property(proxy)
+        if namespace.keydesc in key_list:
+            raise StratisCliNameConflictError("key", namespace.keydesc)
+
+        ((changed, existing_modified), return_code, message) = add_update_key(
+            proxy,
+            namespace.keydesc,
+            namespace.capture_key,
+            keyfile_path=namespace.keyfile_path,
+        )
+
+        if return_code != StratisdErrors.OK:  # pragma: no cover
```

Comment:
You should be able to remove the "no cover" pragma here, since the engine should reliably return an error if the key is too long.